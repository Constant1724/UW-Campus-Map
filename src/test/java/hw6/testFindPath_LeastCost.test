# START_TEST
# NAME LeastCost
# test a graph can be loaded and correctly find the path from A to B, with least
# cost, as cost being the alphabetical ordering of edges and labels..

# Create a graph
CreateGraph graph1

# AddNode
AddNode graph1 ValidNode1
AddNode graph1 ValidNode2
AddNode graph1 ValidNode3
AddNode graph1 ValidNode4
AddNode graph1 ValidNode5
AddNode graph1 ValidNode6

AddEdge graph1 ValidNode1 ValidNode2 label12b
AddEdge graph1 ValidNode1 ValidNode2 label12a

AddEdge graph1 ValidNode1 ValidNode3 label13

AddEdge graph1 ValidNode1 ValidNode4 label14b
AddEdge graph1 ValidNode1 ValidNode4 label14a

AddEdge graph1 ValidNode2 ValidNode5 label25

AddEdge graph1 ValidNode5 ValidNode6 label56

AddEdge graph1 ValidNode2 ValidNode6 label26b
AddEdge graph1 ValidNode2 ValidNode6 label26a

AddEdge graph1 ValidNode4 ValidNode6 label46b
AddEdge graph1 ValidNode4 ValidNode6 label46a

FindPath graph1 ValidNode1 ValidNode6
# END_TEST