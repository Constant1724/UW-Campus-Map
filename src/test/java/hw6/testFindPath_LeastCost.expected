# START_TEST
# NAME LeastCost
# test a graph can be loaded and correctly find the path from A to B, with least
# cost, as cost being the alphabetical ordering of edges and labels..

# Create a graph
created graph graph1

# AddNode
added node ValidNode1 to graph1
added node ValidNode2 to graph1
added node ValidNode3 to graph1
added node ValidNode4 to graph1
added node ValidNode5 to graph1
added node ValidNode6 to graph1

added edge label12b from ValidNode1 to ValidNode2 in graph1
added edge label12a from ValidNode1 to ValidNode2 in graph1

added edge label13 from ValidNode1 to ValidNode3 in graph1

added edge label14b from ValidNode1 to ValidNode4 in graph1
added edge label14a from ValidNode1 to ValidNode4 in graph1

added edge label25 from ValidNode2 to ValidNode5 in graph1

added edge label56 from ValidNode5 to ValidNode6 in graph1

added edge label26b from ValidNode2 to ValidNode6 in graph1
added edge label26a from ValidNode2 to ValidNode6 in graph1

added edge label46b from ValidNode4 to ValidNode6 in graph1
added edge label46a from ValidNode4 to ValidNode6 in graph1

path from ValidNode1 to ValidNode6:
ValidNode1 to ValidNode2 via label12a
ValidNode2 to ValidNode6 via label26a
# END_TEST